src/socket/tcp.rs

```diff
@@ -230,6 +230,8 @@ pub struct TcpSocket<'a> {
     /// The speculative remote window size.
     /// I.e. the actual remote window size minus the count of in-flight octets.
     remote_win_len:  usize,
+    /// The receive window scaling factor for remotes which support RFC-1323, None if unsupported.
+    remote_win_scale: Option<u8>,
```

Comment:
Do we really need an `Option` here? The `None` case seems identical to `Some(0)` in every respect.

Reply:
Nit (here and elsewhere): we write RFC numbers as `RFC NNNN`, not `RFC-NNNN`.

Reply:
I will gladly fix the RFC numbers.

The reason for an option type is twofold. First, it is how phase 2 will handle the scaling window in my current prototype. Second, it is important in the case of listening sockets.

Our TCP should only respond with the window scaling option in the SYN-ACK if the client's SYN contained it. On line 1405 where that logic is applied, the only way to know what the original SYN contained is to check the field set on line 1032 when the packet was parsed. That value must represent the difference between "option not set" and "option set to multiplier of 1". Hence, values of ``None`` versus ``Some(0)``.